src/openforms/formio/utils.py

```diff
@@ -140,7 +140,7 @@ def get_component_default_value(component) -> Optional[Any]:
     return default_value
 
 
-def mimetype_allowed(mime_type: str, allowed_mime_types: List[str]) -> bool:
+def mimetype_allowed(mime_type: str, allowed_mime_types: Container[str]) -> bool:
```

Comment:
it's not wrong to use `Container` - but it does expand the API a bit beyond intended usage - the allowed mimetypes typically comes from the formio configuration which uses a list (since it's a JSON datatype, where tuples/sets/dicts don't exist).

Duplicate entries aren't really a problem for the implementation, so I'd revert this back and use lists instead (zen of python - preferably one way to do something :wink: )

Reply:
I'm in love with `PathLike` and programming to an interface/protocol instead of an implementation in general. If an algorithm just iterates, use an `Iterable` or `Sequence`, if it just checks with `in`, a `Container` is what the algorithm expects and actually uses. It's all over the Django code too. Almost everywhere you can interchangeably use lists or tuples everywhere. "Explicit is better than implicit", this explicitly communicates what part of the `List` api is actually expected. "... Although that way may not be obvious at first unless you're Dutch." :wink:

Reply:
but it does limit your implementation for future refactoring - stretching it up to `Container` does remove your "freedom" to alter implementation details and call `List` specific interface (e.g. access the `allowed_mime_types.index(<item>)` to include this information in a validation error.

In general I try to be a specific and constrained as possible, because it's easier to relax constraints than tighten them in the future.